components/Offer.tsx

```diff
@@ -0,0 +1,494 @@
+        const address = offer.offerBundle.nfts.contractAddresses[i];
+
+        if (
+          nft.asset_contract.address.toUpperCase() === address.toUpperCase() &&
+          nft.token_id == id.toString()
+        ) {
+          return true;
+        }
+      }
+
+      return false;
+    });
+
+    const askNFTs = data.filter((nft) => {
+      for (let i = 0; i < offer.askBundle.nfts.ids.length; i++) {
+        const id = offer.askBundle.nfts.ids[i].toNumber();
+        const address = offer.askBundle.nfts.contractAddresses[i];
+
+        if (
+          nft.asset_contract.address.toUpperCase() === address.toUpperCase() &&
+          nft.token_id == id.toString()
+        ) {
+          return true;
+        }
+      }
+
+      return false;
+    });
+
+    return [askNFTs, offerNFTs];
+  };
+
+  const grabTokenMetaData = async (offer): Promise<[any[], any[]]> => {
+    const offerAddresses = offer.offerBundle.tokens.contractAddresses;
+    let offerTokens = [];
+
+    if (offerAddresses.length) {
+      offerTokens = await Web3Api.token.getTokenMetadata({
+        chain: process.env.NEXT_PUBLIC_CHAIN,
+        addresses: offerAddresses,
+      } as any);
+    }
+
+    const askAddresses = offer.askBundle.tokens.contractAddresses;
+    let askTokens = [];
+
+    if (askAddresses.length) {
+      askTokens = await Web3Api.token.getTokenMetadata({
+        chain: process.env.NEXT_PUBLIC_CHAIN,
+        addresses: askAddresses,
+      } as any);
+    }
+
+    // To get tokens to display in token container, lol this is hacky
+    // to-do huge token balances that will overflow js numbers
+    offerTokens = offerTokens.map((token, i) => {
+      console.log(offer.offerBundle.tokens.amounts[i].toNumber());
+
+      return {
+        ...token,
+        decimals: "0",
+        balance: offer.offerBundle.tokens.amounts[i].toNumber().toString(),
+      };
+    });
+
+    askTokens = askTokens.map((token, i) => {
+      return {
+        ...token,
+        decimals: "0",
+        balance: offer.askBundle.tokens.amounts[i].toNumber().toString(),
+      };
+    });
+
+    return [askTokens, offerTokens];
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        setLoadingMetaData(true);
+
+        const [askNFTs, offerNFTs] = await grabNFTMetaData(offer);
+        const [askTokens, offerTokens] = await grabTokenMetaData(offer);
+
+        setAskNFTs(askNFTs);
+        setOfferNFTs(offerNFTs);
+        setOfferTokens(offerTokens);
+        setAskTokens(askTokens);
+
+        setLoadingMetaData(false);
+      } catch (error) {
+        toast({
+          title: "An error occured",
+          description: error.message,
+          status: "error",
+        });
+      }
+    };
+
+    getData();
+  }, [offer]);
+
+  useEffect(() => {
+    const resolveENS = async () => {
+      const ens = await addressToEns(otherAddress);
+      if (ens) {
+        setOtherAddress(ens);
+      }
+    };
+
+    resolveENS();
```

Comment:
On line 305 you're doing a `trimAddress` on the other address, but we probably shouldn't trim the ENS name.  Can we do the `trimAddress` in this block if it doesn't find an ENS name?

Reply:
done